### ISSUE TITLE
react-native link occasionally errs with "str.replace is not a function", no way to debug

### ISSUE THREAD

**Issue by j2kun** [facebook/react-native#12940]
### Description

I have an xcode workspace set up with nonstandard configurations, and adding a new react-native ios dependency sometimes fails with `str.replace is not a function`. I have no idea how to debug this, as the linking command doesn't provide a `--verbose` option, or anything similar.

E.g.

```
react-native link react-native-image-crop-picker
rnpm-install info Android module react-native-image-crop-picker is already linked 
rnpm-install info Linking react-native-image-crop-picker ios dependency 
rnpm-install ERR! It seems something went wrong while linking. Error: str.replace is not a function 
Please file an issue here: https://github.com/facebook/react-native/issues 

str.replace is not a function
```

I cannot reproduce this issue from a fresh project install, and I have no idea where to look for the problem.

```
react-native-cli: 2.0.1
react-native: 0.40.0
```

**Comment by dantman** [comment]
I fixed one instance of this type of error in #13373 where `.replace` was called on `packageName` when it can be undefined.

If you want to dig into the source of an error in `link` you can dump the error to the console from here:
https://github.com/facebook/react-native/blob/0a712f5335d1bf04c4ff93a79b14c9e23dbd5831/local-cli/link/link.js#L176-L182

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.
